### ISSUE TITLE
what is the reason that epoch affect the mAP

### ISSUE THREAD

**Issue by alicera** [ultralytics/yolov5#10963]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I set the --epoch 100 and --epoch 300 to train the dataset

I found the the mAP of --epoch 100 is higher than the mAP of --epoch 300 when they are all at 100 epoch.

why? I expect the the mAP is the same with +-1% range

**Comment by glenn-jocher** [comment]
@alicera the mAP fluctuation between epochs can be influenced by various factors including learning rate, optimizer, data augmentation, and class imbalance. It's common to observe such variations and experimenting with hyperparameters like learning rate decay, optimizer selection, and augmentation policies may assist in stabilizing mAP across epochs. 🚀
